### ISSUE TITLE
Think about React folder structure

### ISSUE THREAD

**Issue by Sergih28** [Sergih28/groc#24]
Look for examples of how to organize the folder structure in a React project.

Keep in mind that there has to be a place for tests, and unit and e2e test should be in separate folders.

Also keep in mind that we are going to use Storybook.

Right now you can see in the [app readme](https://github.com/Sergih28/groc/blob/main/app/README.md#-project-structure) our basic folder structure given by astro.

**Comment by Sergih28** [comment]
@ehdlg Post here the examples or your proposals for the folder structure here, so we can discus on them and then decide.

An example would be the [react atomic design](https://github.com/danilowoz/react-atomic-design).

**Comment by ehdlg** [comment]
Here are my proposals:

First proposal:

```text
/
├── src/
│   ├── components/
│   ├── layouts/
│   ├── pages/
│   ├── tests/            // Folder for all tests
│   │   ├── unit/         // Folder for unit tests
│   │   │   
│   │   └── e2e/          // Folder for end-to-end tests
│   │      
│   └── stories/          // Folder for Storybook 
└──
```

Second proposal:

```text
/
├── src/
│   ├── components/
│   ├── layouts/
│   ├── pages/
│   ├── unitTests/        // Folder for unit tests
│   │   └── ...          
│   ├── e2eTests/         // Folder for end-to-end tests
│   │   └── ...          
│   └── stories/          // Folder for Storybook 
└──
```

**Comment by Sergih28** [comment]
@ehdlg 

> ```
> /
> ├── src/
> │   ├── components/
> │   ├── layouts/
> │   ├── pages/
> │   ├── tests/            // Folder for all tests
> │   │   ├── unit/         // Folder for unit tests
> │   │   │   
> │   │   └── e2e/          // Folder for end-to-end tests
> │   │      
> │   └── stories/          // Folder for Storybook 
> └──
> ```

I like more this approach, but it needs more description inside the components folder.

**Comment by Sergih28** [comment]
Reminder for when reviewing this.

Add a folder for models (for typescript types and interfaces)
The tests folder goes outside src.

**Comment by Sergih28** [comment]
Merged documentation in #142
